### PR TITLE
Automatically Quote Identifiers with Spaces in Their Names

### PR DESCRIPTION
**Pull request by derekrprice**
|      Q       |   A
|------------- | -----------
| Type         | bug
| BC Break     | no
| Fixed issues | #2979 

#### Summary
Automatically quote identifiers with spaces in their names.

I know that automatically quoting identifiers is frowned upon, for the reasons described in #146, but `AbstractAsset::getquotedName` already makes an exceptions for keywords, since not quoting those would break the generated SQL anyhow.  In the same way, spaces in column names and other identifiers break the generated SQL, so I think this is justified.

### CODE REVIEW

**Comment by SenseException** [comment]
Thank you. Can you please add tests for your additional code?

**Comment by derekrprice** [comment]
I've added tests.  I'd like to see this on the 2.11.x branch too.  I expect that I should create a second pull request for that?

**Comment by greg0ire** [comment]
IMO you should click the EDIT button and change the base to 2.11.x, and then run `git rebase --onto origin/2.11.x origin/master master && git push --force`. It will be a bit confusing because your branch is named `master`, but that should not be an issue.

**Comment by derekrprice** [comment]
I created pull request #3876 instead since I already had this change on a 2.11.x derivative branch, I couldn't edit the source branch, and there are merge conflicts from 2.11.x -> master that I had already resolved anyhow.

**Comment by derekrprice** [comment]
Should I close this PR in favor of the other?

**Comment by greg0ire** [comment]
I think you should indeed. We are switching to a merge up strategy, so your changes should land on master in the end, just no through this PR.

File: tests/Doctrine/Tests/DBAL/Platforms/AbstractPlatformTestCase.php [hunk 1]
```diff
@@ -629,17 +629,30 @@ public function testKeywordList() : void
...
-    public function testQuotedColumnInPrimaryKeyPropagation() : void
+    public function testAutoQuotedColumnInPrimaryKeyPropagation(string $identifier) : void
     {
         $table = new Table('`quoted`');
-        $table->addColumn('create', 'string', ['length' => 255]);
-        $table->setPrimaryKey(['create']);
+        $table->addColumn($identifier, 'string', ['length' => 255]);
+        $table->setPrimaryKey([$identifier]);
 
+        $expected = preg_replace('/create/', $identifier, $this->getQuotedColumnInPrimaryKeySQL());
```
Comment (greg0ire) [inline tests/Doctrine/Tests/DBAL/Platforms/AbstractPlatformTestCase.php]: Not a huge fan of this… I would add a new argument defaulting to `create` to `getQuotedColumnInPrimaryKeySQL`.

**Pull request closed without merging**